src/serializer/ResidualHeapSerializer.js

```diff
@@ -1091,10 +1124,128 @@ export class ResidualHeapSerializer {
+            name = t.identifier(propertyNameOrSymbol);
+          } else {
+            name = this.serializeValue(propertyNameOrSymbol);
+          }
+          invariant(classProtoId !== undefined);
+          this.emitter.emit(
+            t.expressionStatement(t.assignmentExpression("=", t.memberExpression(classProtoId, name), methodId))
+          );
+        }
+      }
+    };
 
+    let serializeClassProperty = (propertyNameOrSymbol, propertyValue) => {
+      // we handle the prototype via class syntax
+      if (propertyNameOrSymbol === "prototype") {
+        this.serializedValues.add(propertyValue);
+      } else if (propertyValue instanceof ECMAScriptSourceFunctionValue && propertyValue.$HomeObject === classFunc) {
+        serializeClassMethod(propertyNameOrSymbol, propertyValue);
+      } else {
+        let prop = classFunc.properties.get(propertyNameOrSymbol);
+        invariant(prop);
+        propertiesToSerialize.set(propertyNameOrSymbol, prop);
+      }
+    };
+
+    // find the all the properties on the class that we need to serialize
+    for (let [propertyName, method] of classFunc.properties) {
+      if (
+        !this.residualHeapInspector.canIgnoreProperty(classFunc, propertyName) &&
+        !ClassProprtiesToIgnore.has(propertyName)
+      ) {
+        withDescriptorValue(propertyName, method.descriptor, serializeClassProperty);
+      }
+    }
+    // pass in the properties and set it so we don't serialize the prototype
     undelay();
+    this._emitObjectProperties(classFunc, propertiesToSerialize, undefined, undefined, true);
 
-    this._emitObjectProperties(val);
+    // handle non-symbol properties
+    for (let [propertyName, method] of classPrototype.properties) {
+      withDescriptorValue(propertyName, method.descriptor, serializeClassMethod);
+    }
+    // handle symbol properties
+    for (let [symbol, method] of classPrototype.symbols) {
+      withDescriptorValue(symbol, method.descriptor, serializeClassMethod);
+    }
+    // assign the AST method key node for the "constructor"
+    classMethodInstance.classMethodKeyNode = t.identifier("constructor");
+
+    // handle class inheritance
+    if (!(classFunc.$Prototype instanceof NativeFunctionValue)) {
+      let proto = classFunc.$Prototype;
+      classMethodInstance.classSuperNode = this.serializeValue(classFunc.$Prototype);
+      if (proto.$HomeObject instanceof ObjectValue) {
+        this.serializedValues.add(proto.$HomeObject);
+      }
+    }
+  }
+
+  _serializeClassMethod(key: string | SymbolValue, methodFunc: ECMAScriptSourceFunctionValue): void {
+    let classMethodInstance = this.residualClassMethodInstances.get(methodFunc);
+
+    invariant(classMethodInstance !== undefined);
+    if (typeof key === "string") {
+      classMethodInstance.classMethodKeyNode = t.identifier(key);
+      // as we know the method name is a string again, we can remove the computed status
+      classMethodInstance.classMethodComputed = false;
```

Comment:
How do we know this? If this is a string but an invalid identifier like `class Foo { ["foo bar"]() { } }` then it still must be computed.

Or if I rename this:

```js
class Foo {
  bar() {
  }
}
Foo.prototype['foo bar'] = Foo.prototype.bar;
delete Foo.prototype.bar;
```